controllers/characters.js

```diff
@@ -0,0 +1,91 @@
+const models = require('../models')
+
+const getAllCharacters = async (request, response) => {
+  try {
+    const characters = await models.characters.findAll()
+
+    return response.send(characters)
+  } catch (error) {
+    return response.status(500).send('Unable to retrieve characters, try again')
+  }
+}
+
+const getCharacterById = async (request, response) => {
+  try {
+    const { identifier } = request.params
+    const characterMatch = await models.characters.findOne({
+      include: [
+        { model: models.sagas },
+        { model: models.races }
+      ],
+      where: {
+        [models.Op.or]: [
+          { id: identifier },
+          { name: { [models.Op.like]: `%${identifier.toLowerCase()}%` } }
+        ]
+      }
+    })
+
+
+    return characterMatch
+      ? response.send(characterMatch)
+      : response.status(404).send(`Unable to find an character matching: ${identifier}`)
+  } catch (error) {
+    return response.status(500).send('Unable to retrieve character, try again')
+  }
+}
+
+const saveNewCharacter = async (request, response) => {
+  try {
+    const {
+      name,
+      race,
+      type,
+      transformation,
+      sagas
+    } = request.body
+
+    if (!name || !race || !type || !transformation || !sagas) {
+      return response
+        .status(400)
+        .send('You are missing at least one of the required fields: name, race, type, transformation, and/or sagas')
+    }
+
+    const existingRace = await models.races.findOne({ where: { name: race } })
```

Comment:
What if the race didn't already exist. You should have error handling here if you expect it to.